src/cplscheme/config/CouplingSchemeConfiguration.cpp

```diff
@@ -307,33 +301,8 @@ void CouplingSchemeConfiguration::xmlEndTagCallback(
   PRECICE_TRACE(tag.getFullName());
   if (tag.getNamespace() == TAG) {
     if (_config.type == VALUE_SERIAL_EXPLICIT) {
-
-      //Check the waveform order of both participants in the explicit coupling
-      if (_experimental) {
-        const auto first  = _config.participants[0];
-        const auto second = _config.participants[1];
-
-        auto first_participant = _participantConfig->getParticipant(first);
-        for (const auto &dataContext : first_participant->readDataContexts()) {
-          const int usedOrder = dataContext.getInterpolationOrder();
-          // The first participants waveform order has to be 0 for serial explicit coupling
-          int allowedOrder = 0;
-          if (usedOrder != allowedOrder) {
-            PRECICE_ERROR(
-                "You configured <read-data name=\"{}\" mesh=\"{}\" waveform-order=\"{}\" />, but for the serial explicit coupling scheme only a maximum waveform-order of \"{}\" is allowed for the first participant.",
-                dataContext.getDataName(), dataContext.getMeshName(), usedOrder, allowedOrder);
-          }
-        }
-        auto second_participant = _participantConfig->getParticipant(second);
-        for (const auto &dataContext : second_participant->readDataContexts()) {
-          const int usedOrder = dataContext.getInterpolationOrder();
-          if (usedOrder < 0) {
-            PRECICE_ERROR(
-                "You configured <read-data name=\"{}\" mesh=\"{}\" waveform-order=\"{}\" />, but for the serial explicit coupling scheme the waveform-order must be non-negative for the second participant.",
-                dataContext.getDataName(), dataContext.getMeshName(), usedOrder);
-          }
-        }
-      }
+      int maxAllowedOrder = 1; // explicit coupling schemes do not allow higher-order waveform iteration
+      checkWaveformOrderReadData(maxAllowedOrder);
```

Comment:
We don't really need these fine-grained checks anymore. For most explicit schemes, `waveform-order="1"` will be used automatically, but it will result in a zeroth order interpolation, because there is no data available for `WINDOW_END`, but only for `WINDOW_START` when we enter the window. Exception: The second participant in a serial-explicit scheme. Here, a first order interpolant is created. `tests/serial/time/explicit/serial-coupling/ReadWriteScalarDataWithWaveform.xml` corresponds to this setup.

Reply:
We could still print warnings for parallel-explicit and first participant in serial-explicit, right?

Reply:
If we use third order as default, we would always warn about the default in this case. This looks odd to me.